tests/tests/worm-xattr-unit.sh

```diff
@@ -0,0 +1,49 @@
+t_require_commands touch rm setfattr
+
+echo "== test creation"
+touch "$T_D0/file-1"
+SECS=$(date '+%s')
+NSECS=$(date '+%N')
+DELAY=10
+EXP=$(( $SECS + $DELAY ))
+setfattr -n scoutfs.worm.v1_expiration -v $EXP.$NSECS "$T_D0/file-1"
+
+echo "== test value is set correctly"
+getfattr --absolute-names -d -m - "$T_D0/file-1" | grep $EXP.$NSECS &> /dev/null
+if [ $? == 0 ]; then
+    echo "== getfattr matches setfattr"
+fi
+
+echo "== test deletion of xattr 1 sec after setting"
+sleep 1
+setfattr -x scoutfs.worm.v1_expiration "$T_D0/file-1"
+
+echo "== test setting xattr to a previous time after setting"
+setfattr -n scoutfs.worm.v1_expiration -v $SECS.$NSECS "$T_D0/file-1"
```

Comment:
I'd add a test section with expected failures from bad value formats and add a whole ton of nonsense.   "a.b" "......", anything you can think of.   The totl test has some decent examples of it trying to mess up the final three values separated by dots.